presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduPartitioningHandle.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kudu;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.log.Logger;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.connector.ConnectorPartitioningHandle;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.lang.String.format;
+
+public class KuduPartitioningHandle
+        implements ConnectorPartitioningHandle
+{
+    private static final Logger log = Logger.get(KuduPartitioningHandle.class);
+
+    private final String table;
+    private final int bucketCount;
+    private final String schema;
+    private final List<ColumnHandle> bucketColumnIds;
+
+    @JsonCreator
+    public KuduPartitioningHandle(
+            @JsonProperty("schema") String schema,
+            @JsonProperty("table") String table,
+            @JsonProperty("bucketCount") int bucketCount,
+            @JsonProperty("bucketColumnIds") List<ColumnHandle> bucketColumnIds)
+    {
+        this.schema = schema;
+        this.table = table;
+        this.bucketCount = bucketCount;
+        this.bucketColumnIds = bucketColumnIds;
+    }
+
+    @JsonProperty
+    public String getSchema()
+    {
+        return schema;
+    }
+
+    @JsonProperty
+    public String getTable()
+    {
+        return table;
+    }
+
+    @JsonProperty
+    public int getBucketCount()
+    {
+        return bucketCount;
+    }
+
+    @JsonProperty
+    public List<ColumnHandle> getBucketColumnIds()
+    {
+        return bucketColumnIds;
+    }
+
+    @Override
+    public String toString()
+    {
+        return format("schema=%s, table=%s, buckets=%d", schema, table, bucketCount);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        KuduPartitioningHandle that = (KuduPartitioningHandle) o;
+        return bucketCount == that.bucketCount;
```

Comment:
We can check the schemaName and tableName right ?

Reply:
Sorry we cannot check schemaName or tableName, because in join operation, presto will call this KuduPartitioningHandle#equals method to check if these two tables can do join, so if we add name check here, join query will not be able to execute in GROUPED_EXECUTION mode.  :(